lib/rails_multitenant/global_context_registry/current.rb

```diff
@@ -34,8 +34,9 @@ def current!
           current || raise("No current #{name} set")
         end
 
-        def clear_current!
+        def clear_current!(cycle_detector = Set.new)
           GlobalContextRegistry.delete(current_registry_obj)
+          __clear_dependents!(cycle_detector)
```

Comment:
I think something like `cleared` would make more sense than `cycle_detector`. Also, I think it would make a bit more sense to `<< self` in the `clear_current!` method.
```suggestion
        def clear_current!(cleared = Set.new)
          GlobalContextRegistry.delete(current_registry_obj)
          cleared << self
          __clear_dependents!(cleared)
```

Reply:
I renamed to `cleared`. The `<< self` needs to happen before we clear any dependents, otherwise when you do `A.current = A.new` you can end up clearing the value you're trying to set (if there's something down the chain that references back to `A`.